### PR TITLE
gpio ll wip

### PR DESCRIPTION
**Pull request by msemegen**
Changes related to GPIO LL implementation

### CODE REVIEW

File: various.hpp [hunk 1]
```diff
@@ -22,7 +22,11 @@ struct various : private non_constructible
         return static_cast<Type_t>(std::numeric_limits<std::underlying_type_t<Type_t>>::max());
     }
 
-    // TODO: replace with std::to_underlying() when we migrate to cpp23 and gcc11+
```
Comment (kaczirosan) [inline various.hpp:25]: Restore this comment

**Review by kaczirosan — DISMISSED** [review]

File: bit.hpp [hunk 2]
```diff
@@ -15,30 +15,34 @@ namespace xmcu {
 class bit : private non_constructible
 {
 public:
-    template<typename Register_t> constexpr static bool is(Register_t a_register, uint8_t a_index)
+    template<typename Register_t, typename Index_t>
+    [[nodiscard]] constexpr static bool is(Register_t register_a, Index_t index_a)
```
Comment (kaczirosan) [inline bit.hpp]: index_a -> a_index or change coding standard

**Review by kaczirosan — CHANGES_REQUESTED** [review]

File: bit.hpp (hunk 2 above)
Comment (msemegen) [reply bit.hpp]: yes, copy/paste from ng_hal

File: various.hpp (hunk 1 above)
Comment (msemegen) [reply various.hpp:25]: yup

File: bit.hpp (hunk 2 above)
Comment (msemegen) [reply bit.hpp]: done

**Review by kaczirosan — APPROVED** [review]

**Pull request merged**